profiler/src/main/java/eu/clarin/switchboard/profiler/xml/TeiProfiler.java

```diff
@@ -0,0 +1,78 @@
+    public final static String XMLNAME_ROOT_TEI = "TEI";
+    public final static String XMLNAME_ROOT_TEI2 = "TEI.2";
+    public final static String XMLNAME_ROOT_TEI_CORPUS = "teiCorpus";
+
+    public final static String MEDIATYPE_TEI = "application/tei+xml";
+
+    public final static String MEDIATYPE_TEI_DTA = "application/tei+xml;format-variant=tei-dta";
+    public final static String SCHEMALOCATION_DTA_RELAXNG = "http://www.deutschestextarchiv.de/basisformat.rng";
+
+    public final static String MEDIATYPE_TEI_CORPUS = "application/tei+xml;format-variant=teiCorpus";
+
+//    public final static String MEDIATYPE_TEI_ISOSPOKEN = "application/tei+xml;format-variant=tei-iso-spoken";
+//    public final static String MEDIATYPE_TEI_CLANCHA = "application/xml;format-variant=clan-cha";
+//    public final static String MEDIATYPE_TEI_EXMARALDAAEXB = "application/xml;format-variant=exmaralda-exb";
+//    public final static String MEDIATYPE_TEI_FOLKERFLN = "application/xml;format-variant=folker-fln";
+//    public final static String MEDIATYPE_TEI_TRANSCRIBERTRS = "application/xml;format-variant=transcriber-trs";
+
+    XMLInputFactory xmlInputFactory;
+
+    public TeiProfiler() {
+        this(XMLInputFactory.newInstance());
+    }
+
+    public TeiProfiler(XMLInputFactory factory) {
+        this.xmlInputFactory = factory;
+    }
+
+    @Override
+    public List<Profile> profile(File file) throws IOException, ProfilingException {
+        XmlUtils.XmlFeatures xmlFeatures;
+        Profile.Builder profileBuilder = Profile.builder().certain();
+
+        XMLEventReader xmlReader = XmlUtils.newReader(xmlInputFactory, file);
+        try {
+            xmlFeatures = XmlUtils.goAfterRoot(xmlReader);
+            if (xmlFeatures.schemaRelaxNG != null &&
+                    xmlFeatures.schemaRelaxNG.equalsIgnoreCase(SCHEMALOCATION_DTA_RELAXNG)) {
+                profileBuilder.mediaType(MEDIATYPE_TEI_DTA);
+            } else if (XMLNAME_ROOT_TEI.equals(xmlFeatures.rootName.getLocalPart())) {
+                profileBuilder.mediaType(MEDIATYPE_TEI);
+            } else if (XMLNAME_ROOT_TEI2.equals(xmlFeatures.rootName.getLocalPart())) {
+                profileBuilder.mediaType(MEDIATYPE_TEI);
+                String lang = xmlFeatures.rootAttributes.get(new QName("lang"));
+                if (lang != null) {
+                    String lang3 = LanguageCode.iso639_1to639_3(lang);
+                    if (lang3 != null) {
+                        lang = lang3;
+                    }
+                    if (LanguageCode.isIso639_3(lang)) {
+                        profileBuilder.language(lang);
+                    }
+                }
+            } else if (XMLNAME_ROOT_TEI_CORPUS.equals(xmlFeatures.rootName.getLocalPart())) {
+                profileBuilder.mediaType(MEDIATYPE_TEI_CORPUS);
+            }
+        } finally {
+            XmlUtils.close(xmlReader);
+        }
+        return Collections.singletonList(profileBuilder.build());
```

Comment:
I think the return statement can be moved into the try{} block to ensure closing of reader even in (unlikely?) case of an unchecked exception in the profile building or singleton list creation.

Reply:
I think the `xmlReader` is already closed at that point, in line 70.

Reply:
yes, you're right, never mind! I shouldn't be writing comments to code during meetings 😬